lib/matplotlib/backends/_backend_tk.py

```diff
@@ -431,6 +431,16 @@ def _get_toolbar(self):
         return toolbar
 
     def resize(self, width, height):
+        max_size = 1_409_023  # experimentally determined on xorg 1.20.8
+        if (width > max_size or height > max_size):
+            raise ValueError(
+                'You have requested to resize the '
+                f'Tk window to ({width}, {height}), one of which '
+                f'is bigger than {max_size}.  At larger sizes xorg will '
```

Comment:
that message will make no sense on non-linux

Reply:
made the check linux only.